packages/modelence/src/auth/signup.ts

```diff
@@ -71,38 +59,19 @@ export async function handleSignupWithPassword(args: Args, { user, connectionInf
     }
   });
 
-  if (getEmailConfig().provider) {
-    const emailProvider = getEmailConfig().provider;
-    const baseUrl = process.env.MODELENCE_SITE_URL || connectionInfo?.baseUrl;
-
-    // Generate verification token
-    const verificationToken = randomBytes(32).toString('hex');
-    const expiresAt = new Date(Date.now() + time.hours(24));
-
-    // Store token in database
-    await emailVerificationTokensCollection.insertOne({
-      userId: result.insertedId,
-      email,
-      token: verificationToken,
-      createdAt: new Date(),
-      expiresAt,
-    });
-    
-    const verifyUrl = `${baseUrl}/api/_internal/auth/verify-email?token=${verificationToken}`;
-    
-    const template = getEmailConfig()?.verification?.template || defaultEmailVerificationTemplate;
-    // TODO: we should have also the name on this step
-    const htmlTemplate = template({ name: '', email, verificationUrl: verifyUrl });
-    const textContent = htmlTemplate.replace(/<[^>]*>/g, '').replace(/\s+/g, ' ').trim();
-    
-    await emailProvider?.sendEmail({
-      to: email,
-      from: getEmailConfig()?.from || 'noreply@modelence.com',
-      subject: getEmailConfig()?.verification?.subject || 'Verify your email address',
-      text: textContent,
-      html: htmlTemplate,
+  if (ip) {
+    await consumeRateLimit({
+      bucket: 'signup',
+      type: 'ip',
+      value: ip,
     });
```

Comment:
I think this should still be in its original position, before inserting the user document? Otherwise we still do the signup, create the user, and only then check the rate limit, which is too late.